File: simplex.java/src/test/java/kr/co/tmon/simplex/sample/actions/MyActionSet.java
package kr.co.tmon.simplex.sample.actions;

import kr.co.tmon.simplex.actions.AbstractActionBinderSet;
import kr.co.tmon.simplex.actions.IActionBinder;
import kr.co.tmon.simplex.reactivex.Unit;

public class MyActionSet extends AbstractActionBinderSet {
	public IActionBinder<SendSignal, Unit, Unit> sendSignal() {
		return getOrAdd("SendSignal", SendSignal.class);
	}
	
	public IActionBinder<GetBoolean, Boolean, Boolean> getBoolean() {
		return getOrAdd("GetBoolean", GetBoolean.class);
	}
		
	public IActionBinder<GetIntegerString, Integer, String> getIntegerString() {
		return getOrAdd("GetIntegerString", GetIntegerString.class);
	}
}

File: Simplex.Demo/Actions/MyActionSet.cs
using System;
using System.Collections.Generic;
using System.Reactive;
using System.Text;
using Tmon.Simplex.Actions;

namespace SimplexDemo.Actions
{
    public class MyActionSet : AbstractActionBinderSet
    {
        public IActionBinder<SendSignal, Unit> SendSignal
            => GetOrAdd<SendSignal, Unit>();

        public IActionBinder<GetBoolean, bool, bool> GetBoolean
            => GetOrAdd<GetBoolean, bool, bool>();

        public IActionBinder<GetIntegerString, int, string> GetInteger
            => GetOrAdd<GetIntegerString, int, string>();

        public SecondActionSet SecondsSet
            => GetOrAdd<SecondActionSet>();
    }

    public class SecondActionSet : AbstractActionBinderSet
    {
        public IActionBinder<GetBoolean, bool, bool> GetBoolean2
            => GetOrAdd<GetBoolean, bool, bool>();
    }
}
